### PR TITLE
CATO-3938: Implementing boxes for 'Directors report: coho question' p…

### PR DESCRIPTION
**Pull request by tmikus**
…age.

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/AC8021.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import uk.gov.hmrc.ct.accounts.frs10x.retriever.Frs10xAccountsBoxRetriever
+import uk.gov.hmrc.ct.box._
+import uk.gov.hmrc.ct.box.retriever.FilingAttributesBoxValueRetriever
+
+case class AC8021(value: Option[Boolean]) extends CtBoxIdentifier(name = "Do you want to file a directors' report to Companies House?") with CtOptionalBoolean with Input with ValidatableBox[Frs10xAccountsBoxRetriever with FilingAttributesBoxValueRetriever] {
+  override def validate(boxRetriever: Frs10xAccountsBoxRetriever with FilingAttributesBoxValueRetriever): Set[CtValidation] = {
+    val coHoFiling = boxRetriever.retrieveCompaniesHouseFiling().value
+    val hmrcFiling = boxRetriever.retrieveHMRCFiling().value
+    val microEntityFiling = boxRetriever.retrieveMicroEntityFiling().value
+    val fileDRToHmrc = boxRetriever.retrieveAC8023().value.contains(true)
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/AC8021.scala]:
```
boxRetriever.retrieveAC8023().orFalse
```

File: src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/AC8021.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    // This field is required if filing non micro-entity Joint or to CoHo only
+    // or when filing Joint micro-entity AND answered "true" to "AC8023".
+    // In the last case, answering "false" disables "Directors report" section - including this question.
+    if (coHoFiling && (!(hmrcFiling && microEntityFiling) || (hmrcFiling && microEntityFiling && fileDRToHmrc)))
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/AC8021.scala]: This is a little hard to understand. I know the comment helps but still scares me.

Comment (tmikus) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/AC8021.scala]: It's just the way that logic works :)

Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/accounts/frs10x/AC8021.scala]: I know. Just wondering if there is a nicer way to express it. Not saying I have an answer, just wondering.

**Pull request merged**